ang/civicase/CaseDetailsPeopleTab.js

```diff
@@ -79,11 +79,24 @@
       if (onlyChecked || $scope[tab + 'SelectionMode'] === 'checked') {
         return _.collect(_.filter($scope[tab], {checked: true}), idField);
       } else if ($scope[tab + 'SelectionMode'] === 'all') {
-        return _.collect($scope[tab], idField);
+        return _.collect(_.filter($scope[tab], function (el) {
+          return el.contact_id;
+        }), idField);
       }
       return [];
     };
 
+    /**
+     * Filter the records on basis of contact_role key
+     *
+     * @param {Array} records
+     */
+    $scope.getRecordCountWithContacts = function (records) {
```

Comment:
it's difficult to understand what the method does by its name, what you mean by "get record count with contacts"? Records of what?

Reply:
Changed the name to a more sensible one - `getCountOfRolesWithContacts`. Actually the array which is responsible for listing all the available roles may contains roles where there is no contact is assigned. 
Since on the page title and bulk action only applies to the roles where there are some contacts. So we need to have this logic in the place.

PS: I have added a bit more documentation to the function, so that it makes more sense.